server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/WebSocketContract.scala

```diff
@@ -399,42 +402,55 @@ trait WebSocketContract {
                  |}""".stripMargin))
 
             ws.receive()
-              .map { case t: Text => {
-                println(t.payload)
+              .map { case t: Text =>
+                responses.add(t.payload)
                 t.payload
               }
-            }
             ws.receive()
-              .map { case t: Text => {
-                println(t.payload)
+              .map { case t: Text =>
+                responses.add(t.payload)
                 t.payload
               }
-              }
             ws.receive()
-              .map { case t: Text => {
-                println(t.payload)
+              .map { case t: Text =>
+                responses.add(t.payload)
                 t.payload
               }
-              }
         })
         .send(backend)
         .body
 
-    /*
-    TODO find how to record several responses
-    Here is what we receive:
-
-    ```
-    {"@type":"StateChange","changed":{"29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6":{"Mailbox":"2880cfd4-6c01-4dd2-a84b-e4771d070883"}}}
-    {"@type":"StateChange","changed":{"29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6":{"Email":"aa633bf3-8f31-474d-9473-99d383a77e84"}}}
-    {"@type":"Response","requestId":"req-36","sessionState":"2c9f1b12-b35a-43e6-9af2-0106fb53a943","methodResponses":[["Email/set",{"accountId":"29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6","oldState":"2c9f1b12-b35a-43e6-9af2-0106fb53a943","newState":"aa633bf3-8f31-474d-9473-99d383a77e84","created":{"aaaaaa":{"id":"1","blobId":"1","threadId":"1","size":341}}},"c1"]]}
-    ```
-
-     The asserts needs to be:
-
-      - out of order
-      - and we need to ensure that is the latest state
-     */
+    val jmapGuiceProbe: JmapGuiceProbe = server.getProbe(classOf[JmapGuiceProbe])
+    val emailState: String = jmapGuiceProbe.getLatestEmailState(accountId).getValue.toString
+    val mailboxState: String = jmapGuiceProbe.getLatestMailboxState(accountId).getValue.toString
+
+    val mailboxStateChange: String =
+      s"""{
+        |  "@type":"StateChange",
+        |  "changed":{
+        |    "$ACCOUNT_ID":{
+        |      "Mailbox":"$mailboxState"
+        |    }
+        |  }
+        |}"""
+        .stripMargin
+        .filter(_ >= ' ')
+        .replace(" ", "")
```

Comment:
a bit ugly but that's all i can come up with for now

Reply:
I think we can one-line the string